### PR TITLE
Feature/arc 412 meemoobeheer leeszalen module

### PR DESCRIPTION
**Pull request by BavoVdg**
[ARC-412](https://meemoo.atlassian.net/browse/ARC-412)
- Fetch spaces from db
- Implement actions
- Move pages to next.js pages folder
- Fix active navigation border clipping through dropdown items

### CODE REVIEW

**Comment by bertyhell** [comment]
@BavoVdg kan je release/v0.11.0 nog eens in deze branch mergen?

File: src/modules/admin/const/Requests.const.tsx [hunk 1]
```diff
@@ -26,30 +26,30 @@ export const requestStatusFilters = (): TabProps[] => {
...
 		},
 		{
 			id: VisitStatus.APPROVED,
-			label: i18n.t('modules/cp/const/requests___goedgekeurd'),
+			label: i18n?.t('modules/admin/const/requests___goedgekeurd'),
 		},
 		{
 			id: VisitStatus.DENIED,
-			label: i18n.t('modules/cp/const/requests___geweigerd'),
+			label: i18n?.t('modules/admin/const/requests___geweigerd'),
```
Comment (bertyhell) [inline src/modules/admin/const/Requests.const.tsx:41]: no need to add the ? behind i18n when you use the new helper:
```typescript
import { i18n } from '@shared/helpers/i18n';
```

File: src/modules/admin/const/Requests.const.tsx [hunk 2]
```diff
@@ -26,30 +26,30 @@ export const requestStatusFilters = (): TabProps[] => {
...
 		},
 	];
 };
 
 export const RequestTableColumns = (): Column<Visit>[] => [
 	{
-		Header: i18n.t('modules/admin/reading-rooms/pages/requests/requests___leeszaal'),
+		Header: i18n.t('modules/admin/const/requests___leeszaal') || '',
```
Comment (bertyhell) [inline src/modules/admin/const/Requests.const.tsx:48]: no need to add a fallback when you use the new helper:
```typescript
import { i18n } from '@shared/helpers/i18n';

Header: i18n.t('modules/admin/const/requests___leeszaal'), // Will default to Leeszaal
```

File: src/modules/admin/const/Routing.const.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	id: string;
+	label: string;
+	href: string;
+	children?: () => AdminNavigationLink[];
+}
+
+export const ADMIN_NAVIGATION_LINKS = (): AdminNavigationLink[] => [
+	{
+		id: 'spaces-admin',
+		label: i18n?.t('modules/admin/const/routing___leeszalenbeheer') ?? '',
```
Comment (bertyhell) [inline src/modules/admin/const/Routing.const.tsx:13]: remove all ? and defaults for all translations

File: src/modules/admin/layouts/AdminLayout/AdminLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+				})),
+			})),
+		[asPath]
+	);
+
+	const { t } = useTranslation();
+
+	useEffect(() => {
+		dispatch(setShowZendesk(false));
+	}, [dispatch]);
```
Comment (bertyhell) [inline src/modules/admin/layouts/AdminLayout/AdminLayout.tsx:50]: PREFERENCE: i keep all hooks and useEffect code at the top of the component
then the functions
and then the render functions and final return statement

File: src/modules/navigation/components/Navigation/Navigation.consts.tsx [hunk 5]
```diff
@@ -193,7 +193,7 @@ export const getNavigationItemsLeft = (
 		],
 	},
 	{
-		node: renderLink('Admin', '/admin/leeszalenbeheer/aanvragen', {
+		node: renderLink('Admin', '', {
```
Comment (bertyhell) [inline src/modules/navigation/components/Navigation/Navigation.consts.tsx:196]: For admin they asked not to add the dropdown menu during the last demo. Just show the admin link.
```suggestion
		node: renderLink('Admin', '/admin/leeszalenbeheer/leeszalen', {
```

File: src/pages/admin/leeszalenbeheer/leeszalen/index.tsx [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+		});
+	};
+
+	const updateRoomStatus = (roomId: string, status: ReadingRoomStatus) => {
+		VistorSpaceService.update(roomId, {
+			status: status,
+		})
+			.catch(onFailedRequest)
+			.then((response) => {
+				if (response === undefined) {
```
Comment (bertyhell) [inline src/pages/admin/leeszalenbeheer/leeszalen/index.tsx:89]: Might be safer to also check for null:
```suggestion
				if (!response) {
```

File: src/pages/admin/leeszalenbeheer/leeszalen/index.tsx [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+												className="u-mt-16 u-mb-16"
+												count={ReadingRoomsOverviewTablePageSize}
+												start={
+													Math.max(0, filters.page - 1) *
+													ReadingRoomsOverviewTablePageSize
+												}
+												total={readingRooms.total || 0}
+												onPageChange={(pageZeroBased) => {
+													gotoPage(pageZeroBased);
+													// setSelected(null);
```
Comment (bertyhell) [inline src/pages/admin/leeszalenbeheer/leeszalen/index.tsx:202]: Should this commented line be removed?
```suggestion
```

File: src/pages/admin/leeszalenbeheer/leeszalen/index.tsx [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+						'pages/admin/leeszalenbeheer/leeszalen/index___alle-leeszalen-meta-omscrhijving'
+					)}
+				/>
+			</Head>
+
+			<AdminLayout
+				contentTitle={t('pages/admin/leeszalenbeheer/leeszalen/index___alle-leeszalen')}
+			>
+				<div className="l-container">
+					{!isLoading && readingRooms ? (
```
Comment (bertyhell) [inline src/pages/admin/leeszalenbeheer/leeszalen/index.tsx:133]: PREFERENCE: i prefer to create a render function like this:
```typescript
const renderPageContent() {
	if (isLoading) {
		return <Loading/>
	}
	if (isError) {
		return <p>Failed to load vistor spaces</p>
	}
	if (!readingRooms) {
		return renderEmptyMessage()
	}
	return renderVisitorSpaces();
}

return <>
	<Head>...</Head>
	<AdminLayout>
		{renderPageContent()}
	</AdminLayout>
</>
```

File: src/styles/pages/admin/_reading-rooms.scss [hunk 9]
```diff
@@ -11,4 +11,14 @@ $page: "p-admin-reading-rooms";
...
+
+	&__error {
+		&__error {
+			height: 100%;
+			display: flex;
+			justify-content: center;
+			align-items: center;
+			color: $neutral;
+		}
+	}
```
Comment (bertyhell) [inline src/styles/pages/admin/_reading-rooms.scss:23]: where is this used?

**Review by bertyhell — APPROVED** [review]

**Comment by bertyhell** [comment]
Nog 2 vertalingen die verkeerd zijn:
image
image

**Review by bertyhell — CHANGES_REQUESTED** [review]
see global remark translations

File: src/modules/navigation/components/Navigation/Navigation.consts.tsx (hunk 5 above)
Comment (ian-emsens-shd) [reply src/modules/navigation/components/Navigation/Navigation.consts.tsx:196]: Staat dat ergens gedocumenteerd? Ontbreekt precies op https://meemoo.atlassian.net/wiki/spaces/HA2/pages/3448504321/2022-04-14+demo+sprint+9+deel+1+BZT

File: src/modules/reading-room/services/visitor-space/vistor-space.service.ts [hunk 10]
```diff
@@ -11,7 +12,9 @@ export class VistorSpaceService {
 	public static async getAll(
 		searchInput = '',
 		page = 0,
-		size = 20
+		size = 20,
+		orderProp: ReadingRoomOrderProps = ReadingRoomOrderProps.CreatedAt,
+		orderDirection: OrderDirection = OrderDirection.desc
```
Comment (ian-emsens-shd) [inline src/modules/reading-room/services/visitor-space/vistor-space.service.ts:17]: Als default waarde toevoegen aan query param config op relevante pagina's, niet in service.

File: src/modules/reading-room/services/visitor-space/vistor-space.service.ts [hunk 11]
```diff
@@ -58,12 +63,13 @@ export class VistorSpaceService {
 		const formData = new FormData();
 
 		// Set form data
-		formData.append('color', values.color ?? '');
-		formData.append('image', values.image ?? '');
+		values.color && formData.append('color', values.color ?? '');
+		values.image && formData.append('image', values.image ?? '');
```
Comment (ian-emsens-shd) [inline src/modules/reading-room/services/visitor-space/vistor-space.service.ts:67]: Overbodige fallback?
```suggestion
		values.color && formData.append('color', values.color);
		values.image && formData.append('image', values.image);
```

**Review by ian-emsens-shd — CHANGES_REQUESTED** [review]
In general;
- Capitalisatie bij vertalingen
- Vermijd `@typescript-eslint/no-explicit-any` (technical debt)

File: src/styles/pages/admin/_reading-rooms.scss (hunk 9 above)
Comment (BavoVdg) [reply src/styles/pages/admin/_reading-rooms.scss:23]: Error styling in upcoming changes

File: src/modules/navigation/components/Navigation/Navigation.consts.tsx (hunk 5 above)
Comment (bertyhell) [reply src/modules/navigation/components/Navigation/Navigation.consts.tsx]: Nee blijkbaar niet. Ik zet het op de PO sync om dat vast te leggen: https://meemoo.atlassian.net/wiki/spaces/HA2/pages/3463512065/2022-04-28+PO+sync

**Review by bertyhell — APPROVED** [review]

**Pull request merged**